packages/core/src/sdk/shipping/useShippingSimulation.ts

```diff
@@ -109,8 +109,15 @@ export const useShippingSimulation = (shippingItem: ProductShippingInfo) => {
   const { country, postalCode: sessionPostalCode } = useSession()
   const { postalCode: shippingPostalCode } = input
   const shippingPostalCodeRef = useRef(shippingPostalCode)
+  /**
+   * Internally useEffect compares the dependencies by reference - it's a problem when there is an object.
+   * Stringifying the object to compare it by value avoids calling useEffect multiple times
+   * when there is no change in the dependencies.
+   */
+  const shippingItemJson = JSON.stringify(shippingItem)
```

Comment:
hmm, what about using a `useMemo()` in the origin of the `shippingItem` construction (in the `ProductDetails` section) and passing it as prop? I think it solves the problem without using this stringify solution.

```js
  const productShippingInfo = useMemo(() => {
    return {
      id,
      quantity,
      seller: seller.identifier,
    }
  }, [id, quantity, seller.identifier])
```